source/linker.cpp

```diff
@@ -0,0 +1,1098 @@
+// Copyright (c) 2017 Pierre Moreau
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "spirv/1.2/spirv.hpp11"
+
+#include <cassert>
+#include <cstring>
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "diagnostic.h"
+#include "table.h"
+
+
+namespace {
+
+using spirv_word = uint32_t;
+
+spirv_word GetWord(const uint32_t* binary, size_t word_offset) {
```

Comment:
I will keep that in mind and add comments to all new functions I create, as well as to older ones that are missing one.